### PR TITLE
Bug 1644433 - Implement UI/UX decisions

### PR DESCRIPTION
**Pull request by esanuandra**

### CODE REVIEW

**Comment by ionutgoldan** [comment]
Found a bug.

**STR**
1. Pick an alert summary with no perf tags.
2. Click on `Add tags`.
3. Check one of the tags.
4. Close the dialog without saving anything.
5. Click on `Add tags` again.

Expected results:
The checked tag isn't checked anymore.

What we actually get:
The checked tag remained checked.

File: ui/perfherder/alerts/TagsModal.jsx [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+import { getData } from '../../helpers/http';
+import { getApiUrl } from '../../helpers/url';
+import { endpoints } from '../constants';
+
+export default class TagsModal extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      tags: [],
+      initialTags: [],
```
Comment (ionutgoldan) [inline ui/perfherder/alerts/TagsModal.jsx:24]: These two var names are ambiguous. It's not immediately obvious what they're holding.
I actually thought `tags` holds all the available tags, while `initialTags` hold those of the summary.
But after reading the code, I discovered it's actually the other way around.

File: ui/perfherder/alerts/AlertHeader.jsx [hunk 2]
```diff
@@ -37,6 +37,18 @@ const AlertHeader = ({
     ? `Bug ${alertSummary.bug_number}`
     : '';
 
+  const listTags = () => {
+    const performanceTags = alertSummary.performance_tags || [];
+
+    if (performanceTags.length > 0) {
+      return performanceTags.map((tag) => (
+        <Badge key={tag} className="mr-2">
```
[the remark this answers is not in the document]
Comment (esanuandra) [reply ui/perfherder/alerts/AlertHeader.jsx:45]: The pill badges seem to be a better choice. Thank you!

File: ui/perfherder/App.jsx [hunk 3]
```diff
@@ -87,6 +91,7 @@ class App extends React.Component {
                     user={user}
                     projects={projects}
                     frameworks={frameworks}
+                    performanceTags={performanceTags}
```
Comment (ionutgoldan) [inline ui/perfherder/App.jsx:94]: I think we've forgotten to guard these routes by uninitialized `performanceTags`.
Currently, they're guarded by this condition:
```javascript
{projects.length > 0 && frameworks.length > 0 && (
  <route configs>
}
```

File: ui/perfherder/Validation.jsx [hunk 4]
```diff
@@ -33,6 +33,7 @@ const withValidation = ({ requiredParams }, verifyRevisions = true) => (
         selectedTimeRange: null,
         framework: null,
         validationComplete: false,
+        performanceTags: [],
```
Comment (ionutgoldan) [inline ui/perfherder/Validation.jsx:36]: I'm not sure this is necessary. At least not at this moment.

File: ui/perfherder/alerts/AlertHeader.jsx [hunk 5]
```diff
@@ -121,6 +122,11 @@ const AlertHeader = ({
           />
         </Col>
       </Row>
+      <Row>
+        <Col className="p-0" xs="auto">
```
Comment (ionutgoldan) [inline ui/perfherder/alerts/AlertHeader.jsx:126]: Could we guard this column's creation with a truthy `performance_tags`, like we did with `bugNumber` above?

File: ui/perfherder/Validation.jsx (hunk 4 above)
Comment (esanuandra) [reply ui/perfherder/Validation.jsx:36]: Indeed, this was not necessary. Thank you!

File: ui/perfherder/alerts/TagsModal.jsx [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+      <Modal isOpen={showModal}>
+        <ModalHeader toggle={this.toggleModal}>Alert Tags</ModalHeader>
+        <Form>
+          <ModalBody>
+            <Label for="performanceTags" />
+            <FormGroup>{this.listTags()}</FormGroup>
+          </ModalBody>
+          <ModalFooter>
+            <Button
+              color="secondary"
```
Comment (sarah-clements) [inline ui/perfherder/alerts/TagsModal.jsx:95]: You should use "darker-secondary" instead: https://treeherder.readthedocs.io/accessibility.html#colors

File: ui/perfherder/alerts/StatusDropdown.jsx [hunk 7]
```diff
@@ -274,6 +289,10 @@ export default class StatusDropdown extends React.Component {
                     Mark as fixed
                   </DropdownItem>
                 )}
+
+                <DropdownItem onClick={() => this.toggle('showTagsModal')}>
```
Comment (sarah-clements) [inline ui/perfherder/alerts/StatusDropdown.jsx:293]: I think this needs to have a `tag=a` attribute per our docs on accessibility changes. The `DropdownItems` above might also need to have that added as well.

File: ui/perfherder/alerts/StatusDropdown.jsx [hunk 8]
```diff
@@ -274,6 +289,10 @@ export default class StatusDropdown extends React.Component {
...
+                  {!alertSummaryActiveTags.length ? 'Add tags' : 'Edit tags'}
```
Comment (sarah-clements) [inline ui/perfherder/alerts/StatusDropdown.jsx:294]: I'm curious what tags are - it seems a bit vague. Are they labels? Would a tooltip with a more verbose explanation make it clearer?

File: ui/perfherder/constants.js [hunk 9]
```diff
@@ -107,3 +107,12 @@ export const compareDefaultTimeRange = {
   value: 86400 * 2,
   text: 'Last 2 days',
 };
+
+export const performanceTagsTips = {
+  harness:
+    'patches that updated that harness and caused improvements/regressions',
```
Comment (ionutgoldan) [inline ui/perfherder/constants.js:113]: Small typo for the 2nd `that`.

File: ui/perfherder/constants.js [hunk 10]
```diff
@@ -107,3 +107,12 @@ export const compareDefaultTimeRange = {
...
+  infra:
+    'improvements/regressions caused by infra changes (changes not related to repository code)',
```
Comment (ionutgoldan) [inline ui/perfherder/constants.js:115]: Could we title case all these tool tip messages?

Comment (ionutgoldan) [reply ui/perfherder/constants.js:115]: Also, I think we could use the singular for `patches`. Because an alert summaries has a single culprit patch.

**Pull request merged**